mhs-reference-implementation/mhs/handler/tests/test_client_request_handler.py

```diff
@@ -6,33 +6,43 @@
 from mhs.handler.client_request_handler import ClientRequestHandler
 from mhs.sender.sender import UnknownInteractionError
 
+MOCK_UUID = "5BB171D4-53B2-4986-90CF-428BE6D157F5"
+INTERACTION_NAME = "interaction"
+REQUEST_BODY = "A request"
 
-class TestAsyncResponseHandler(AsyncHTTPTestCase):
+
+class TestClientRequestHandler(AsyncHTTPTestCase):
 
     def get_app(self):
         self.sender = Mock()
         return Application([
-            (r"/.*", ClientRequestHandler, dict(sender=self.sender))
+            (r"/.*", ClientRequestHandler, dict(sender=self.sender, callbacks={}, async_timeout=1))
         ])
 
-    def test_post(self):
-        interaction_name = "interaction"
-        request_body = "A request"
+    def test_post_synchronous_message(self):
```

Comment:
Is there a test missing here to check the post of the asynchronous message. Which I think at the moment adds to the callable dictionary. Do you think this would be valuable to have?

Reply:
Yes there is, and I do think it would be valuable, but this is the case that would test the asynchronous messaging pattern, as I mentioned in the description. Since this requires the asynchronous response to be returned in order for the post method to return I think this is going to require a more involved integration test, which I felt was beyond the scope of this task.

Reply:
I see is there not a "simple" unit test that can be added that will prove the callback dictionary is updated when it is determined that the request is of type `async`. I say "simple" I have no idea how to do it myself or how involved that would be. If you feel the integration test would best cover this then happy to leave this as is?

Reply:
As discussed offline, I haven't come up with a straightforward approach, so I feel that the integration test would be the best way to do this.